internal/identity/gitlab.go

```diff
@@ -0,0 +1,127 @@
+package identity // import "github.com/pomerium/pomerium/internal/identity"
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	oidc "github.com/pomerium/go-oidc"
+	"github.com/pomerium/pomerium/internal/httputil"
+	"github.com/pomerium/pomerium/internal/sessions"
+	"github.com/pomerium/pomerium/internal/version"
+	"golang.org/x/oauth2"
+)
+
+const (
+	defaultGitLabProviderURL = "https://gitlab.com"
+	revokeURL                = "https://gitlab.com/oauth/revoke"
+	oidcURL                  = "/oauth/userinfo"
+)
+
+// GitLabProvider is an implementation of the OAuth Provider
+type GitLabProvider struct {
+	*Provider
+	Subject       string `json:"sub"`
+	Profile       string `json:"profile"`
+	Email         string `json:"email"`
+	EmailVerified bool   `json:"email_verified"`
+
+	RevokeURL string `json:"revocation_endpoint"`
+}
+
+// NewGitLabProvider returns a new GitLabProvider.
+// https://www.pomerium.io/docs/identity-providers.html#gitlab-active-directory
```

Comment:
dead url

Reply:
@desimone I should have updated that part when I updated the docs. Nice catch!